frontstage/views/surveys/upload_survey.py

```diff
@@ -34,37 +34,50 @@ def upload_survey(session):
     party_controller.is_respondent_enrolled(party_id, business_party_id, survey_short_name)
 
     # Get the uploaded file
-    upload_file = request.files['file']
-    upload_filename = upload_file.filename
+
+    # The variable name has '_raw' so that it can be called in the file size checking function,
+    # and not interfere with the actual upload function
+
+    upload_file_raw = request.files['file']
+    upload_filename = upload_file_raw.filename
+
     upload_file = {
-        'file': (upload_filename, upload_file.stream, upload_file.mimetype, {'Expires': 0})
+        'file': (upload_filename, upload_file_raw.stream, upload_file_raw.mimetype, {'Expires': 0})
     }
 
-    try:
-        # Upload the file to the collection instrument service
-        collection_instrument_controller.upload_collection_instrument(upload_file, case_id, party_id)
-    except CiUploadError as ex:
-        if ".xlsx format" in ex.error_message:
-            error_info = "type"
-        elif "50 characters" in ex.error_message:
-            error_info = "charLimit"
-        elif "File too large" in ex.error_message:
-            error_info = 'size'
-        else:
-            logger.error('Unexpected error message returned from collection instrument service',
-                         status=ex.status_code,
-                         error_message=ex.error_message,
-                         party_id=party_id,
-                         case_id=case_id)
-            error_info = "unexpected"
+    # Checking if the file size is great than 5999 bytes
+    if collection_instrument_controller.check_collection_instrument_size(upload_file_raw, party_id, case_id):
```

Comment:
The logic in this bit is really weird now so I think it should be rewritten (though maybe as part of another PR after this one so too much isn't going on at once)
The validation for the file is really weird.  So it looks like previously ALL the validation was done in the collection-instrument service.  We hand it whatever crap the user gives to us, we pass it to the service and it tells us if something is wrong.  Now we do a bit of validation on our side, they do the rest of the validation on their side. 

Now that I write this.  I think the collection instrument service should also get the same treatment?  If something else wanted to post a collection instrument, there'd be no safeguards to make sure that it's not a 0 byte file, even though we know that 0 byte files are wrong

Reply:
Though after talking to JT, what's been done here is fine as is.  If we want to improve it later on we'll just do it down the line.  Just need to fix the other comments I have and it should be good to go.
I tested it in my sandbox environment (mostly to play with the github actions) and it works well.